src/main/java/org/opentripplanner/model/calendar/ServiceDateInterval.java

```diff
@@ -0,0 +1,116 @@
+package org.opentripplanner.model.calendar;
+
+import java.util.Objects;
+
+
+/**
+ * Value object witch represent an service date interval from a starting date
+ * until an end date. Both start and end is inclusive.
+ *
+ * The {@code start} must be equals or before the {@code end} to form a valid
+ * period.
+ *
+ * {@code null} is used to represent an unbounded interval. One or both the of the {@code start} and
+ * {@code end} can be {@code null} (unbounded).
+ */
+public final class ServiceDateInterval {
+
+    /**
+     * The unbounded values will represent an unbounded value internally in this class.
+     * They are converted to {@code null} when exported outside the class. This little
+     * trick make internal logic a bit simpler, since there is no need for {@code null} checks.
+     */
+    private static final ServiceDate UNBOUNDED_START = new ServiceDate(0, 1, 1);
+    private static final ServiceDate UNBOUNDED_END = new ServiceDate(9999, 12, 31);
+    private static final ServiceDateInterval UNBOUNDED = new ServiceDateInterval(null, null);
+
+    private final ServiceDate start;
+    private final ServiceDate end;
+
+
+    public ServiceDateInterval(ServiceDate start, ServiceDate end) {
+        this.start = start == null ? UNBOUNDED_START : start;
+        this.end = end == null ? UNBOUNDED_END : end;
+
+        // Guarantee that the start is before or equal the end.
+        if(this.end.isBefore(this.start)) {
+            throw new IllegalArgumentException(
+                    "Invalid interval, the end " + end + " is before the start " +  start
+            );
+        }
+    }
+
+    /**
+     * Return a interval with start or end unbounded ({@code null}).
+     */
+    public static ServiceDateInterval unbounded() {
+        return UNBOUNDED;
+    }
+
+    public boolean isUnbounded() {
+        return start.equals(UNBOUNDED_START) && end.equals(UNBOUNDED_END);
+    }
+
+    public ServiceDate getStart() {
+        return start == UNBOUNDED_START ? null : start;
+    }
+
+    public ServiceDate getEnd() {
+        return end == UNBOUNDED_END ? null : end;
```

Comment:
I made moved the constants to the `ServiceDate` class and made them public. Also made them exposed to the outsid, by not converting to `null`. The new names for these are `MIN_DATE` and `MAX_DATE` since the `UNBOUNDED` only make sence in the context of interval.